src/probnum/filtsmooth/particlefiltsmooth/_particle_filter.py

```diff
@@ -106,6 +107,8 @@ def filter(self, dataset, times):
             Posterior distribution of the filtered output.
         """
 
+        dataset, times = regression_problem.observations, regression_problem.locations
```

Comment:
do you think it would make sense to rename `dataset` and `times`? Or would this make it less verbose?

Reply:
If you ask for my opinion: I don't find it necessary. But I can certainly change it if you prefer.
However, as it is now, it does align with the parameter naming of lower-level methods, as e.g.
```python
filtrv, *_ = self.update(
            data=dataset[0],
            rv=self.initrv,
            time=times[0],
            _linearise_at=_linearise_update_at,
        )
```

Reply:
okay keep it